marquez_airflow/dag.py

```diff
@@ -125,6 +126,27 @@ def _timed_log_message(self, start_time):
             f'marquez_namespace={self.marquez_namespace} ' \
             f'duration_ms={(self._now_ms() - start_time)}'
 
+    def _handle_task_state(self, kwargs, dagrun, task_instances):
+        for ti in task_instances:
+            self.log.debug(f"{ti}")
+            job_name = self._marquez_job_name_from_ti(ti)
+            session = kwargs.get('session')
+
+            marquez_job_run_ids = self._job_id_mapping.pop(
+                job_name, dagrun.run_id, session)
```

Comment:
This modifies the _job_id_mapping object and will cause a run to not be reported correctly in a later section of code. Moving this PR to Draft until it can be addressed.